spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -569,9 +571,17 @@ public ConsumerRecords<K, V> receive(Collection<TopicPartitionOffset> requested,
 		Properties props = oneOnly();
 		Map<TopicPartition, List<ConsumerRecord<K, V>>> records = new LinkedHashMap<>();
 		try (Consumer<K, V> consumer = this.consumerFactory.createConsumer(null, null, null, props)) {
+			AtomicInteger index = new AtomicInteger(0);
 			requested.forEach(tpo -> {
+				if (tpo.getOffset() == null) {
+					throw new KafkaException(String.format("Offset supplied in TopicPartitionOffset [%d] is null", index.get()));
```

Comment:
I think we just can use a `tpo` in the exception message to indicate exactly which one is wrong.
And therefore no need in that `index` variable to track.

Reply:
done

Reply:
@artembilan @garyrussell  please could you review when you get a chance?
Thanks.